`domainRegex` does not match k8s or docker DNS names

in Docker or Kubernetes a DNS name can just be a single name, an example for `docker-compose.yml`:

```
services:
  core-lightning:
    build:
      context: .
      dockerfile: Dockerfile

  core-lightning-application:
    build:
      context: .
      dockerfile: core-lightning-application.dockerfile
```

will allow you to resolve just `core-lightning` inside the `core-lightning-application` container, example:

```# dig core-lightning

; <<>> DiG 9.16.42-Debian <<>> core-lightning
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 48634
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;core-lightning.			IN	A

;; ANSWER SECTION:
core-lightning.		600	IN	A	172.18.0.6

;; Query time: 2 msec
;; SERVER: 127.0.0.11#53(127.0.0.11)
;; WHEN: Fri Jul 21 18:26:06 UTC 2023
;; MSG SIZE  rcvd: 62
```

but unfortunately `lnmessage` does not like just `core-lightning`, there is an error:

```core-lightning-application_1  | > cln-application@0.0.3 start
core-lightning-application_1  | > npm run serve -w cln-application-backend
core-lightning-application_1  |
core-lightning-application_1  |
core-lightning-application_1  | > cln-application-backend@0.0.3 serve
core-lightning-application_1  | > node ./dist/server.js
core-lightning-application_1  |
core-lightning-application_1  | {"level":"error","message":"\tFailed to read rune for Commando connection: {}","timestamp":"2023-07-21 18:24:55:2455"}
core-lightning-application_1  | file:///app/node_modules/lnmessage/dist/validation.js:10
core-lightning-application_1  |         throw new Error(`${ip} is not a valid IP or DNS address`);
core-lightning-application_1  |               ^
core-lightning-application_1  |
core-lightning-application_1  | Error: core-lightning is not a valid IP or DNS address
core-lightning-application_1  |     at validateInit (file:///app/node_modules/lnmessage/dist/validation.js:10:15)
core-lightning-application_1  |     at new LnMessage (file:///app/node_modules/lnmessage/dist/index.js:71:9)
core-lightning-application_1  |     at new LightningService (file:///app/apps/backend/dist/service/lightning.service.js:20:34)
core-lightning-application_1  |     at file:///app/apps/backend/dist/service/lightning.service.js:55:26
core-lightning-application_1  |     at ModuleJob.run (node:internal/modules/esm/module_job:194:25)
core-lightning-application_1  |
core-lightning-application_1  | Node.js v18.16.0
core-lightning-application_1  | npm ERR! Lifecycle script `serve` failed with error:
core-lightning-application_1  | npm ERR! Error: command failed
core-lightning-application_1  | npm ERR!   in workspace: cln-application-backend@0.0.3
core-lightning-application_1  | npm ERR!   at location: /app/apps/backend```

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "lnmessage",
-  "version": "0.2.2",
+  "version": "0.2.2-0.0.1",
   "description": "Talk to Lightning nodes from your browser",
   "main": "dist/index.js",
   "type": "module",

--- a/src/validation.ts
+++ b/src/validation.ts
@@ -10,10 +10,10 @@ export function validateInit(options: LnWebSocketOptions): void {
 
   const ipRegex = /^((25[0-5]|(2[0-4]|1\d|[1-9]|)\d)(\.(?!$)|$)){4}$/
 
-  const domainRegex =
-    /^((?!-))(xn--)?[a-z0-9][a-z0-9-_]{0,61}[a-z0-9]{0,1}\.(xn--)?([a-z0-9-]{1,61}|[a-z0-9-]{1,30}\.[a-z]{2,})$/
+  const dnsRegex =
+    /^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9-]*[A-Za-z0-9])$/
 
-  if (!ip || (!ip.match(ipRegex) && !ip.match(domainRegex) && ip !== 'localhost')) {
+  if (!ip || (!ip.match(ipRegex) && !ip.match(dnsRegex) && ip !== 'localhost')) {
     throw new Error(`${ip} is not a valid IP or DNS address`)
   }
 

